src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -209,42 +214,95 @@ private Claim readCase(User user, String caseId) {
         );
     }
 
-    private List<CaseDetails> search(User user, Map<String, Object> searchString) {
+
+    private List<CaseDetails> search(User user, Map<String, String> searchString) {
+        return search(user, searchString, 1, new ArrayList<>(), null);
+    }
+
+    private List<CaseDetails> search(
+        User user,
+        Map<String, String> searchString,
+        Integer page,
+        List<CaseDetails> results,
+        Integer numOfPages
+    ) {
+        Map<String, String> searchCriteria = new HashMap<>(searchString);
+        searchCriteria.put("page", page.toString());
 
         String serviceAuthToken = this.authTokenGenerator.generate();
 
+        results.addAll(performSearch(user, searchCriteria, serviceAuthToken));
+
+        if (results.size() > MINIMUM_SIZE_TO_CHECK_FOR_MORE_PAGES) {
+            if (numOfPages == null) {
```

Comment:
why `numOfPages` is `Integer` and not `int`?

Reply:
So I can do a null check here, i.e. numOfPages is unknown in the first instance